Sources/FluidAudio/Diarizer/ANEMemoryOptimizer.swift

```diff
@@ -0,0 +1,264 @@
+import Accelerate
+import CoreML
+import Foundation
+import Metal
+
+/// ANE-optimized memory management for speaker diarization pipeline
+@available(macOS 13.0, iOS 16.0, *)
+public class ANEMemoryOptimizer {
```

Comment:
The code here seems mostly duplicated with the ones in the ASR model, is there a way to reduce the duplicate code? 

Perhaps a parent and extensions for each model type if needed?